index.js

```diff
@@ -57,6 +60,10 @@ export default function fablePlugin(userConfig) {
     pendingChanges: null,
     hotPromiseWithResolvers: null,
     isBuild: false,
+    lastTransformId: null,
+    pendingTransformQueue: [], // Track files being transformed
```

Comment:
While the idea of a queue isn't bad, in practice there will only ever be one pending request at a time.